diff --git a/node_modules/react-native-ble-plx/android/src/main/java/com/bleplx/utils/SafePromise.java b/node_modules/react-native-ble-plx/android/src/main/java/com/bleplx/utils/SafePromise.java
index 0203473..f61d19a 100644
--- a/node_modules/react-native-ble-plx/android/src/main/java/com/bleplx/utils/SafePromise.java
+++ b/node_modules/react-native-ble-plx/android/src/main/java/com/bleplx/utils/SafePromise.java
@@ -22,18 +22,27 @@ public class SafePromise {
 
   public void reject(String code, String message) {
     if (isFinished.compareAndSet(false, true)) {
+      if (code == null || code.isEmpty()) {
+        code = "E_BLE_ERROR";
+      }
       promise.reject(code, message);
     }
   }
 
   public void reject(String code, Throwable e) {
     if (isFinished.compareAndSet(false, true)) {
+      if (code == null || code.isEmpty()) {
+        code = "E_BLE_ERROR";
+      }
       promise.reject(code, e);
     }
   }
 
   public void reject(String code, String message, Throwable e) {
     if (isFinished.compareAndSet(false, true)) {
+      if (code == null || code.isEmpty()) {
+        code = "E_BLE_ERROR";
+      }
       promise.reject(code, message, e);
     }
   }
